### PR TITLE
added .editorconfig and rustfmt.tml

### PR DESCRIPTION
**Pull request by SeniorMars**
Before the project becomes any more bigger, I suggest adding both of these files.

### CODE REVIEW

**Comment by beeb** [comment]
Why remove the cargo cache action?

File: .github/workflows/checks.yml
Comment (nvarner) [inline .github/workflows/checks.yml:1]: What are the advantages of the new CI workflow over the existing one?

File: .github/dependabot.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+version: 2
+updates:
+  - package-ecosystem: github-actions
+    directory: /
+    schedule:
+      interval: daily
+  - package-ecosystem: cargo
+    directory: /
+    schedule:
+      interval: daily
```
Comment (nvarner) [inline .github/dependabot.yml:10]: Can we add an `npm` ecosystem at `addons/vscode` for the VS Code extension?

File: .rustfmt.toml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+use_small_heuristics = "Max"
```
Comment (nvarner) [inline .rustfmt.toml:1]: This looks reasonable, but is there any particular reason to prefer it over the default?

**Review by nvarner — COMMENTED** [review]
I have some questions about the changes, but they look good overall. This will help a lot with the stability of the repo.

File: .github/workflows/checks.yml [hunk 3]
```diff
@@ -1,27 +1,100 @@
...
+          submodules: true
+      - name: Install ${{ matrix.toolchain }}
+        uses: dtolnay/rust-toolchain@master
+        with:
+          toolchain: ${{ matrix.toolchain }}
+          components: clippy
+      - name: cargo clippy
+        uses: actions-rs/clippy-check@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
```
[the remark this answers is not in the document]
Comment (beeb) [reply .github/workflows/checks.yml:48]: Actually never mind, it's likely to be useless in most cases due to https://github.com/actions-rs/clippy-check/issues/2

**Comment by nvarner** [comment]
Closing for now. Please reopen if you're able to address the reviews.

**Pull request closed without merging**